waiter/src/waiter/core.clj

```diff
@@ -1233,11 +1217,12 @@
                                     basic-auth-handler (basic-authentication/wrap-basic-authentication handler router-comm-authenticated?)]
                                 (basic-auth-handler request)))))
    :wrap-secure-request-fn (pc/fnk [[:routines authentication-method-wrapper-fn]
-                                    [:state cors-validator]]
+                                    [:state cors-validator]
+                                    [:settings cors-config]]
                              (fn wrap-secure-request-fn
-                               [request-handler]
-                               (let [handler (-> request-handler
-                                                 (cors/handler cors-validator)
+                               [handler]
+                               (let [handler (-> handler
+                                                 (cors/wrap-cors cors-validator (:max-age cors-config))
```

Comment:
`max-age` is already available to the `cors-validator` via the `cors-config` passed during initialization.

Reply:
Yes, but it doesn't expose it.  We put max-age in `cors-config` because it was a natural place to put the config value, even though its also used to configure the component.